src/core/shared_state/currentMapPosition.ts

```diff
@@ -1,70 +1,77 @@
 import { createAtom } from '~utils/atoms';
 import { getCameraForBbox } from '~utils/map/camera';
 import { currentMapAtom } from './currentMap';
+import type { Map } from 'maplibre-gl';
 
 export type CenterZoomPosition = {
-  // type: 'centerZoom';
   lat: number;
   lng: number;
   zoom: number;
 };
 export type Bbox = [number, number, number, number];
 export type BboxPosition = {
-  // type: 'bbox';
   bbox: Bbox;
 };
 
 export type MapPosition = CenterZoomPosition | BboxPosition;
 
 export type CurrentMapPositionAtomState = MapPosition | null;
 
+// TODO: #20160 update currentMapPositionAtom to reatom v3
 export const currentMapPositionAtom = createAtom(
   {
     setCurrentMapPosition: (mapPosition: { lat: number; lng: number; zoom: number }) =>
       mapPosition,
     setCurrentMapBbox: (mapBbox: Bbox | [[number, number], [number, number]]) => mapBbox,
+    updateCurrentMapPosition: (mapPosition: { lat: number; lng: number; zoom: number }) =>
```

Comment:
@albaranau  `CenterZoomPosition ` There is a simpler and shorter name for this object - `Viewport`

Reply:
Seems like it's better not to rename the type, because actual Viewport is expected to have more info:
https://konturio.slack.com/archives/C02DEKP2MKP/p1731932373198899?thread_ts=1731931161.995599&cid=C02DEKP2MKP

Reply:
Maplibre has "CameraOptions" type 
<img width="815" alt="image" src="https://github.com/user-attachments/assets/14e47103-0177-4045-82d8-b1b4c8fbfef1">